component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```
@@ -103,25 +115,51 @@ public Response getUsers(@Context UriInfo uriInfo,
     offset = offset > 0 ? offset : RestUtils.getOffset(uriInfo);
     limit = limit > 0 ? limit : RestUtils.getLimit(uriInfo);
 
-    ProfileFilter filter = new ProfileFilter();
-    filter.setName(q == null || q.isEmpty() ? "" : q);
+    Identity[] identities;
+    int totalSize = 0;
 
-    ListAccess<Identity> list = CommonsUtils.getService(IdentityManager.class).getIdentitiesByProfileFilter(OrganizationIdentityProvider.NAME, filter, false);
-
-    Identity[] identities = list.load(offset, limit);
+    if (StringUtils.isNotBlank(status)) {
+      String userId;
+      try {
+        userId = ConversationState.getCurrent().getIdentity().getUserId();
+      } catch (Exception e) {
+        return Response.status(HTTPStatus.UNAUTHORIZED).build();
+      }
+      if (StringUtils.isBlank(userId)) {
+        return Response.status(HTTPStatus.UNAUTHORIZED).build();
+      }
+      Space space = null;
+      if (StringUtils.isNotBlank(spaceId)) {
+        space = spaceService.getSpaceById(spaceId);
+      }
+      if (space != null) {
+        identities = getOnlineIdentitiesOfSpace(userId, space, limit);
+      } else {
+        identities = getOnlineIdentities(userId, limit);
```

If spaceId is not null and no space exist with this space, we should return a 404 error IMO.

---

Yes, ok